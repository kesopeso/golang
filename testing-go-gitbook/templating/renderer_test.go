package templating_test

import (
	"bytes"
	"templating"
	"testing"

	approvals "github.com/approvals/go-approval-tests"
)

func TestRenderPost(t *testing.T) {
	post := templating.Post{
		Title:       "Test post",
		Description: "Testing rendering",
		Tags:        []string{"rendering", "tests"},
		Body: `# Body title

This is some body to test the render with

## Subtitle

Another body paragraph.`,
	}

	postRenderer, err := templating.NewPostRenderer()
	if err != nil {
		t.Fatal("error should not occur", err)
	}
	buf := bytes.Buffer{}
	err = postRenderer.RenderPost(&buf, post)
	if err != nil {
		t.Fatal("error should not occur", err)
	}

	approvals.VerifyString(t, buf.String())
}

func TestRenderIndex(t *testing.T) {
	posts := []templating.Post{
		{Title: "Post 1"},
		{Title: "Post 2"},
		{Title: "Post 3"},
	}

	postRenderer, err := templating.NewPostRenderer()
	if err != nil {
		t.Fatal("error should not occur", err)
	}
	buf := bytes.Buffer{}
	err = postRenderer.RenderIndex(&buf, posts)
	if err != nil {
		t.Fatal("error should not occur", err)
	}

	approvals.VerifyString(t, buf.String())
}
