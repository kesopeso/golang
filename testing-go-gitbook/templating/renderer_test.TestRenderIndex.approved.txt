<!DOCTYPE html>
<html lang="en">
    <head>
        <title>My amazing blog!</title>
        <meta charset="UTF-8" />
        <meta name="description" content="Wow, like and subscribe!" />
    </head>
    <body>
        <nav role="navigation">
            <div>
                <h1>My blog</h1>
                <ul>
                    <li><a href="/">home</a></li>
                    <li><a href="/">about</a></li>
                    <li><a href="/">archive</a></li>
                </ul>
            </div>
        </nav>
    </body>
    <main>
<h1>Posts</h1>
<ol><li><a href="/posts/post-1">Post 1</a></li><li><a href="/posts/post-2">Post 2</a></li><li><a href="/posts/post-3">Post 3</a></li></ol>
    </main>
    
    <footer>
        <ul>
            <li><a href="https://x.com">X - Twitter</a></li>
            <li><a href="https://github.com">Github</a></li>
        </ul>
    </footer>
</html>
